File: packages/aws-cdk/lib/init-templates/app/java/src/main/java/com/myorg/HelloConstruct.java
package com.myorg;

import software.amazon.awscdk.core.Construct;
import software.amazon.awscdk.services.iam.IGrantable;
import software.amazon.awscdk.services.s3.Bucket;
import software.amazon.awscdk.services.s3.BucketProps;

import java.util.ArrayList;
import java.util.List;

/**
 * Example of a reusable construct. This one defines N buckets.
 */
public class HelloConstruct extends Construct {
    private final List<Bucket> buckets = new ArrayList<>();

    public HelloConstruct(final Construct parent, final String name, final HelloConstructProps props) {
        super(parent, name);

        for (int i = 0; i < props.getBucketCount(); ++i) {
            buckets.add(new Bucket(this, "Bucket" + String.valueOf(i), BucketProps.builder().build()));
        }
    }

    /**
     * Given an principal, grants it READ access on all buckets.
     * @param grantee The principal (User, Group, Role)
     */
    public void grantRead(final IGrantable grantee) {
        buckets.forEach(b -> b.grantRead(grantee, "*"));
    }
}

File: packages/aws-cdk/lib/init-templates/sample-app/python/hello/hello_construct.py
from aws_cdk import (
    aws_iam as iam,
    aws_s3 as s3,
    core,
)


class HelloConstruct(core.Construct):

    @property
    def buckets(self):
        return tuple(self._buckets)

    def __init__(self, scope: core.Construct, id: str, num_buckets: int) -> None:
        super().__init__(scope, id)
        self._buckets = []
        for i in range(0, num_buckets):
            self._buckets.append(s3.Bucket(self, f"Bucket-{i}"))
    
    def grant_read(self, principal: iam.IPrincipal):
        for b in self.buckets:
            b.grant_read(principal, "*")
